app/Http/Controllers/BuildController.php

```diff
@@ -543,6 +543,45 @@ public function apiViewConfigure(): JsonResponse
         return response()->json(cast_data_for_JSON($response));
     }
 
+    public function apiBuildUpdateGraph(): JsonResponse
+    {
+        // Find previous submissions from this build.
+        $query_result = DB::select('
+            SELECT
+                b.id,
+                b.starttime,
+                bu.nfiles
+            FROM build as b
+            JOIN build2update AS b2u ON b2u.buildid = b.id
+            JOIN buildupdate AS bu ON bu.id = b2u.updateid
+            WHERE
+                b.siteid = ?
+                AND b.type = ?
+                AND b.name = ?
+                AND b.projectid = ?
+                AND b.starttime <= ?
+            ORDER BY b.starttime ASC
+        ', [
+            $this->build->SiteId,
```

Comment:
This seemed to work at first, but now I'm getting a 500 error after clicking on `Show Activity Graph` saying that the build isn't initialized yet.  This is on the build that errors on the update, maybe that is the problem?

```
"error": "Typed property App\\Http\\Controllers\\AbstractBuildController::$build must not be accessed before initialization",
    "code": 500,
    "trace": {
        "message": "Typed property App\\Http\\Controllers\\AbstractBuildController::$build must not be accessed before initialization",
        "exception": "Error",
        "file": "\/cdash\/app\/Http\/Controllers\/BuildController.php",
        "line": 565,
```

Reply:
Good catch!  I must have forgotten to clear my route cache when I tested my changes...

It certainly helps to set the build before trying to access it!